helpers/contracts.js

```diff
@@ -96,8 +99,45 @@ exports.updateVersion = async (contractId, versionId, versionToUpdate) => {
   return contract;
 };
 
+exports.deleteVersion = async (contractId, versionId) => {
+  const contract = await Contract.findOne({ _id: contractId }).lean();
+  if (!contract.versions || contract.versions.length === 0) return null;
+
+  const isLastVersion = contract.versions[contract.versions.length - 1]._id.toHexString() === versionId;
+  if (!isLastVersion) throw Boom.forbidden();
+  const deletedVersion = contract.versions[contract.versions.length - 1];
+
+  if (contract.versions.length > 1) {
+    await Contract.updateOne(
+      { _id: contractId, 'versions._id': versionId },
+      { $pull: { versions: { _id: versionId } } },
+      { autopopulate: false }
+    );
+    await Contract.updateOne(
```

Comment:
Et je me demandais aussi: Ne peut-on pas plus simplement passer par Mongoose, sans `lean` donc, modifier notre contrat, puis `.save()` ?
https://mongoosejs.com/docs/tutorials/lean.html#when-to-use-lean